Adds UDN IR limitation to UDN docs (#92561)
openshift/openshift-docs: modules/nw-udn-limitations.adoc

* *Network policy limitations*: Network policies that enable traffic between namespaces connected to different user-defined primary networks are not effective. These traffic policies do not take effect because there is no connectivity between these isolated networks.

* *Creation and modification limitation*: The `ClusterUserDefinedNetwork` CR and the `UserDefinedNetwork` CR cannot be modified after being created.

> * *Image registry access*: A user-defined network pod cannot currently access the {product-title} image registry. As a result, running pods cannot pull images. Because of this limitation, source-to-image builds do not work in a user-defined network namespace. Additionally, other functions, such as `oc new-app <command>` that are used to create new applications based on the source code in a Git repository, or from an {product-title} template that use source-to-image builds, do not work. This limitation might also affect other `openshift-*.svc` services.

Reviewer: I think it may be easier to understand if we emphasize the general rule about default-network service access and use image registry as an example. So something like

*Default network service access*: A user-defined network pod is isolated from the default network, that means that most of the default network services are not accessible. For example, a user-defined network pod cannot currently access the {product-title} image registry. Because of this limitation, source-to-image builds do not work in a user-defined network namespace. Additionally, other functions, such as `oc new-app <command>` that are used to create new applications based on the source code in a Git repository, or from an {product-title} template that use source-to-image builds, do not work. This limitation might also affect other `openshift-*.svc` services.

I removed `As a result, running pods cannot pull images.` part, because it sounds scary like pods can't pull any images, but I think the next sentence explains it better.

> * *Default network service access*: A user-defined network pod is isolated from the default network, which means that most default network services are inaccessible. For example, a user-defined network pod cannot currently access the {product-title} image registry. Because of this limitation, source-to-image builds do not work in a user-defined network namespace. Additionally, other functions, such as `oc new-app <command>` that are used to create new applications based on the source code in a Git repository, or from an {product-title} template that use source-to-image builds, do not work. This limitation might also affect other `openshift-*.svc` services.

Reviewer 2: ```suggestion
* *Default network service access*: A user-defined network pod is isolated from the default network, which means that most default network services are inaccessible. For example, a user-defined network pod cannot access the {product-title} image registry. Because of this limitation, source-to-image builds do not work in a user-defined network namespace. Additionally, other functions do not work, including functions to create applications based on the source code in a Git repository, such as `oc new-app <command>`, and functions to create applications from an {product-title} template that use source-to-image builds. This limitation might also affect other `openshift-*.svc` services.
```
- If it's important to include "currently" in the second sentence, feel free to ignore my suggestion.
- In the third sentence, I tried to place the subject closer to the verb. If my revision isn't quite right, feel free to tweak it or disregard it.
